kafka-admin/src/main/resources/graphql-schema/kafka-admin.graphql

```diff
@@ -1,11 +1,67 @@
-extend type Query {
-    topic(topicName: String): Topic
-    topicList(filter: String): [Topic]
+type TopicDescription {
+    #id: ID
+    name: String
+
+    """@transient"""
+    partitions: [Partitions]
+    config: TopicConfig
+}
+
+type Partitions {
+    partition: Int
+    replicas: [Replicas]
+}
+
+type Replicas {
+    id: String
+    inSync: Boolean
 }
 
-type Topic {
+type TopicOnlyName {
```

Comment:
No strong opinion but this might be redundant due to the capabilities of the client to make query with selection set containing only name. Making type for it makes things much more complex and also will put constraints on growing API.

This is not comment to fix it etc. I'm just mentioning it from the point of schema good practices as even if someone will request extra fields in topic query they will get null data.

Reply:
I agree. This is still a  WIP and I was expecting (hoping for) a comment like this.

Reply:
Totally agree we don't need this.